Sources/SourceKitLSP/Documentation/DocumentationManager.swift

```diff
@@ -0,0 +1,274 @@
+        documentPathsToConvert: nil,
+        includeRenderReferenceStore: false,
+        documentationBundleLocation: nil,
+        documentationBundleDisplayName: moduleName ?? "Unknown",
+        documentationBundleIdentifier: "unknown",
+        symbolGraphs: symbolGraphs,
+        overridingDocumentationComments: overridingDocumentationComments,
+        emitSymbolSourceFileURIs: false,
+        markupFiles: [],
+        tutorialFiles: [],
+        convertRequestIdentifier: UUID().uuidString
+      ) { convertResponse in
+        switch convertResponse {
+        case .success(let convertResponse):
+          guard let renderNodeData = convertResponse.renderNodes.first else {
+            continuation.resume(throwing: ResponseError.internalError("SwiftDocC did not return any render nodes"))
+            return
+          }
+          guard let renderNode = String(data: renderNodeData, encoding: .utf8) else {
+            continuation.resume(throwing: ResponseError.internalError("Failed to encode render node from SwiftDocC"))
+            return
+          }
+          continuation.resume(returning: .renderNode(renderNode))
+        case .failure(let serverError):
+          continuation.resume(throwing: serverError)
+        }
+      }
+    }
+  }
+}
+
+fileprivate final class DocumentableSymbolFinder: SyntaxAnyVisitor {
+  struct Symbol {
+    let position: AbsolutePosition
+    let documentationComments: [String]
+  }
+
+  private let cursorPosition: AbsolutePosition
+
+  /// Accumulating the result in here.
+  private var result: Symbol? = nil
+
+  private init(_ cursorPosition: AbsolutePosition) {
+    self.cursorPosition = cursorPosition
+    super.init(viewMode: .sourceAccurate)
+  }
+
+  /// Designated entry point for `DocumentableSymbolFinder`.
+  static func find(
+    in nodes: some Sequence<Syntax>,
+    at cursorPosition: AbsolutePosition
+  ) -> Symbol? {
+    let visitor = DocumentableSymbolFinder(cursorPosition)
+    for node in nodes {
+      visitor.walk(node)
+    }
+    return visitor.result
+  }
+
+  private func setResult(node: some SyntaxProtocol, position: AbsolutePosition) {
+    setResult(
+      result: Symbol(
+        position: position,
+        documentationComments: node.leadingTrivia.flatMap { trivia -> [String] in
+          switch trivia {
+          case .docLineComment(let comment):
+            return [String(comment.dropFirst(3).trimmingCharacters(in: .whitespaces))]
+          case .docBlockComment(let comment):
+            return comment.dropFirst(3)
+              .dropLast(2)
+              .split(separator: "\n")
+              .map { String($0).trimmingCharacters(in: .whitespaces) }
+          default:
+            return []
+          }
+        }
+      )
+    )
+  }
+
+  private func setResult(result symbol: Symbol) {
+    if result == nil {
+      result = symbol
+    }
+  }
+
+  private func visitNamedDeclWithMemberBlock(
+    node: some SyntaxProtocol,
```

Comment:
Should this be the following? If so, we also don’t need to pass `name` because we can access it from `node.name`.

```suggestion
    node: some NamedDeclSyntax,
```

Reply:
Good point! Done.